azure-pipelines.yml

```diff
@@ -30,22 +32,59 @@ steps:
 
 - script: |
     set -x
-    docker pull $(docker_path):latest || true
-  displayName: "Pull latest docker image to cache"
+    docker pull $(docker_path):$GIT_BRANCH || true
+    docker pull $(docker_path)-middleman:$GIT_BRANCH || true
+  displayName: "Pull any existing images for this branch to cache"
+  # When this branch is first created, there won't be any images to cache, even though master
```

Comment:
You could pull master always, and pull from the branch conditionnally ex: `[ $GIT_BRANCH != master ] && ...`

Reply:
I think the tricky bit is in the `docker-compose.yml`, it needs to `cache_from` the correct image (I ran into a problem when specifying multiple `cache_from` so that wouldn't work). We could drive that from a env var, maybe, but I think it may be worth it to get what's here in first.